### DISCUSSION TITLE
Help setting up OIDC

### DISCUSSION THREAD

**Discussion by r3yno** [open-webui/open-webui#5559] [Q&A] [answered]
Hello, 
My team is looking to implement Open WebUI into our environment and I was tasked with setting up SSO or SAML for it. However consulting the documentation provided I am completely lost on how to implement it. I see referenced to environment variables needing to be set, however setting them seems to have no noticeable result. We are currently running it in a docker container on a red hat server.

I feel like I am missing a Rosetta stone detail that will make everything make sense, having been at this for a couple days I am still lost on how to implement it. I have put the envs into the bashrc and on profiles.

**Comment by ThomasKubus** [comment]
Hi @r3yno, did you make any progress here? Because I have the same problem :D A colleague of mine helps me to setup SAML and asked me for a Reply URL (Assertion Consumer Service URL) and a Sign On URL and I don't know what to put there

**Reply by justinh-rahb** [reply]
No SAML support here, you won't be getting very far.

**Reply by c-hoffmann** [reply]
I'm in the same boat, looking for a LLM-GUI with SAML support.
@ThomasKubus: Have you found a solution?
@justinh-rahb: Is SAML ruled out for future development? Or can I interpret #4858 and #4883 as SAML coming to open-webui?

**Reply by ThomasKubus** [reply]
not yet sorry :(

**Comment by r3yno** [comment, marked as answer]
@ThomasKubus 
I should have followed up here when I found an answer. What I ended up doing was spinning up a Keycloak server, and used the OIDC protocol to authenticate. Tying my Keycloak server to my org's domain is how I am handling the SSO experience. 

The missing piece of information I needed, was knowing that Docker can only pass environment variables on the run command. Run of course meaning creation as opposed to start, So I talked my team into killing our docker container for Open WebUI, and started it over. 

Once in place, I found it to work very well, and would recommend Keycloak if you need a free open sourced alternative to other OIDC providers.

**Reply by ThomasKubus** [reply]
Thank you for your reply. How exactly do i have to set up OIDC? Is there any helpful documentation online? :)
